### PR TITLE
FT Edit all accommodations

### PR DESCRIPTION
**Pull request by Cheza-Dzabala**
#### What does this PR do?
- Allows a supplier to edit their accommodations

#### Description of Task to be completed?
- A supplier should be able to edit the following:

- The supplier is able to edit a name for his facility.
-The supplier is able to edit a description of his facility.
- The supplier is able to edit the location (i.e address) for his facility.
- The supplier is able to edit an image/images of his/her facility.
- The supplier is able to edit the space available in his facility.
- The supplier is able to edit the cost of his accommodation facility. 
- The supplier is able to edit on services that his/her accommodation facility offers.
- The supplier is able to edit amenities that his facility offers

#### How should this be manually tested?

1. Clone this repository
2. run `npm install`
3. run `npm run dev`
4. Log in with a supplier account
5. Access the accommodations section of the app
6. Locate an accommodation owned by the supplier and hit edit
7. Update details of the accommodation and you will be redirected to the accommodation page

#### Any background context you want to provide?
N /A
#### What are the relevant pivotal tracker stories?
 #170308209
#### Screenshots (if appropriate)
Screen Shot 2019-12-19 at 11 45 45
Screen Shot 2019-12-19 at 11 46 02
Screen Shot 2019-12-19 at 11 46 26
Screen Shot 2019-12-19 at 11 46 50

#### Questions:
N / A

### CODE REVIEW

**Review by nignanthomas — CHANGES_REQUESTED** [review]
@Cheza-Dzabala 
Great job bro

Kindly add an `Edit` button also in `Single Accommodation` for the Supplier.
Also, on mobile view (iPad especially) the textareas are too squeezed as you can see here.
Hope you do not mind working on that.
Screen Shot 2019-12-19 at 15 54 17

**Review by nignanthomas — APPROVED** [review]

**Review by sabin18 — CHANGES_REQUESTED** [review]
@Cheza-Dzabala good job! However, it is better to display images horizontally. Work on that.

File: src/styles/components/editor.scss [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.editor {
+	border: 1px solid #c2c2c2;
```
Comment (nakiwuge) [inline src/styles/components/editor.scss]: Move this color to variables

File: src/styles/components/editor.scss [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+	padding: 10px;
+	height: 50px;
+	background-color: white;
```
Comment (nakiwuge) [inline src/styles/components/editor.scss]: Move this color to variables

File: src/tests/accommodationTests/editAccommodationTest.spec.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  const wrapper = shallow(<EditAccommodations store={store} {...props} />).childAt(0).dive();
+  return wrapper;
+};
+
+describe('Test Request Component', () => {
+
+  it('Should show loading page', async () => {
+    const component = setUp(initialState, {})
+    expect(findByTestAttribute(component, 'loading-page').length).toBe(1);
+
```
Comment (nakiwuge) [inline src/tests/accommodationTests/editAccommodationTest.spec.js]: Remove this line

File: src/tests/accommodationTests/editAccommodationTest.spec.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  it('Should call handleSubmit', async () => {
+    const component = setUp(initialState, {});
+    component.setProps({
+      locations: locations, accommodation: accommodationMocks, GetSingleAccommodation: jest.fn(),
+      getLocations: jest.fn(),
+    })
+    component.setState({ accommodation: accommodationMocks, isLoading: false })
+    component.instance().handleChange({ target: { value: 'this hotel', name: 'name' } });
+    component.instance().updateAccommodation({ target: { ame: 'name', }, preventDefault: jest.fn() });
```
Comment (nakiwuge) [inline src/tests/accommodationTests/editAccommodationTest.spec.js:103]: Add an expect

File: src/views/accommodations/editAccommodation.js [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+    if (images) {
+      Object.values(images).forEach((image) => {
+        formData.append('image', image);
+      });
+    }
+
+    for (const key in data) {
+      if (data[key]) {
+        console.log(key);
```
Comment (nakiwuge) [inline src/views/accommodations/editAccommodation.js]: Remove console

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@Cheza-Dzabala  good job on the implementation. Kindly work on the feedback below
- Move the image update to the bottom of the form it does not look good on the top.
- Add a `cancel` button. This button should redirect the user back to the accommodation they were editing.
- Make the `Edit` button on `viewing single accommodation` a link and move it to the extreme right preferably in line with the breadcrumbs
- Raise your coverage to 90+ and check out the other comments I have left on this pr

**Review by sabin18 — APPROVED** [review]
Good job @Cheza-Dzabala .

**Comment by Cheza-Dzabala** [comment]
> @Cheza-Dzabala good job on the implementation. Kindly work on the feedback below
> 
> * Move the image update to the bottom of the form it does not look good on the top.
> * Add a `cancel` button. This button should redirect the user back to the accommodation they were editing.
> * Make the `Edit` button on `viewing single accommodation` a link and move it to the extreme right preferably in line with the breadcrumbs
> * Raise your coverage to 90+ and check out the other comments I have left on this pr

Noted. 
- Coverage raised. 
- Added Cancel button. 
- Moved the edit link to far-right. 
- Moved image edit to the bottom of the page

File: src/tests/accommodationTests/editAccommodationTest.spec.js (hunk 4 above)
Comment (Cheza-Dzabala) [reply src/tests/accommodationTests/editAccommodationTest.spec.js:103]: Fixed this, not sure why the `outdated` tag isn't pasted on it.

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**